### ISSUE TITLE
"into_<fields>" methods

### ISSUE THREAD

**Issue by mattfbacon** [bluejekyll/trust-dns#1865]
Consider this struct: https://docs.rs/trust-dns-proto/0.22.0/trust_dns_proto/rr/rdata/mx/struct.MX.html

It has an owned Name field, but there is no way to get that out of it; you can only get &Name.

For efficiency purposes, I would like to be able to do `into_exchange` to get Name from MX.

This applies in general to all similar data types.

**Comment by djc** [comment]
Do you actually have a benchmark where this matters?

In general, pull requests for stuff like this are welcome! For this particular type, I wonder if we should just make the fields public instead...

**Comment by mattfbacon** [comment]
I agree that making the fields public would be the best option if possible. I thought they were private for backwards-compatibility or something.

I don't think it makes a huge performance difference. It's just annoying to have to Clone when I would be able to take out the field.

**Comment by bluejekyll** [comment]
Generally, I like keeping internal fields private, like you said @mattfbacon, for backward compatibility concerns. That being said, the RecordData types probably won't change, so maybe it's not a concern there?

**Comment by mattfbacon** [comment]
OK if making the fields public is acceptable then I will make a PR doing that soon.

**Comment by djc** [comment]
Yeah, for `RecordData` fields I think making the fields public is probably reasonable.

**Comment by mattfbacon** [comment]
Were these structs generated automatically somehow?

**Comment by djc** [comment]
Nope!
